device/protocol_mqtt_v1.go

```diff
@@ -270,13 +286,69 @@ func (d *Device) sendMqttV1MessageInternal(astarteInterface interfaces.AstarteIn
 		return err
 	}
 
-	topic := fmt.Sprintf("%s/%s%s", d.getBaseTopic(), astarteInterface.Name, interfacePath)
-	// TODO: Handle this token
-	_ = d.m.Publish(topic, qos, false, doc)
+	// if it's a property, we need to set it
+	if astarteInterface.Type == interfaces.PropertiesType {
+		go d.storeProperty(astarteInterface.Name, interfacePath, doc)
+	}
```

Comment:
It seems you're storing the property before the message is successfully delivered. This would introduce a problem when implementing property deduplication (e.g. the user publishes `42`, you save it, the send fails, the user publishes it again and you don't send it since from your POV the server state already has `42` as value).
Device-owned propertied should be saved to the DB _after_ the successful delivery of the message.